go/pkg/client/cas.go

```diff
@@ -103,6 +104,50 @@ type uploadState struct {
 	cancel  func()
 }
 
+// StatusError is the same as status.Error except it includes the error details in the error message.
+type StatusError struct {
```

Comment:
Maybe have this error outside of `cas.go`? Maybe still under `package client`, but cas.go is large and IIUC this isn't tied at dependant.

Reply:
Yep, agreed. Added status.go file.